app/models/profile.rb

```diff
@@ -100,6 +100,11 @@ def strict_ial2_proofed?
     includes_phone_check?
   end
 
+  def has_proofed_before?
+    threshold = active? ? 2 : 1
+    Profile.where(user_id: user_id).where.not(activated_at: nil).count >= threshold
+  end
```

Comment:
what if instead of checking for a hard count based on active, we did a count, excluding this one's ID?

```suggestion
  def has_proofed_before?
    Profile.where(user_id: user_id).where.not(activated_at: nil, id: self.id).exists?
  end
```

Reply:
I like this. Multiple conditions in a `.not` were getting interpreted non-intuitively so I broke it down to be more explicit, but this is definitely a better way than using a count.